docs/proposal-cylc-rose-installing-rose-configs.md

```diff
@@ -0,0 +1,291 @@
+> undocumented but the answer lies in the code
+> [here](https://github.com/metomi/rose/blob/99ad9e766571212c6c3afdbe3915c75ba05e6c68/metomi/rose/run.py#L149-L155)
+> and
+> [here](https://github.com/metomi/rose/blob/99ad9e766571212c6c3afdbe3915c75ba05e6c68/metomi/rose/config.py#L1280-L1286).
+
+
+`rose suite-run` copies source files into the run directory *excluding* the
+`rose-suite.conf` file.
+
+The parsed Rose config is dumped in the run directory into
+`log/rose-suite.conf` for logging reasons, however, it is not read by Rose
+from this location for any further purpose.
+
+Any configuration set by either (2) or (3) above must be provided
+in any subsequent `rose suite-run` command or it will be lost. This includes
+`rose suite-run --reload` and `rose suite-run --restart` making these
+commands especially dangerous.
+
+## Rose2 & Cylc8
+
+The `rose suite-run` command is better thought of as `rose suite-install --run`.
+By default it runs the suite but can also `--reload` or `--restart`
+on request.
+
+In the new system `rose suite-run` has been replaced by `cylc install`
+which will call out to the new `cylc-rose` plugin for Rose specific
+functionality.
+
+This new command will be responsible for the installation of the Rose
+configuration into the run directory as well as supporting the command line
+options and environment variables that can change the Rose configuration
+(as outlined above).
+
+```
+usage: cylc install [opts]
+
+Plugins:
+  cylc-rose:
+    Opts:
+      -S, --define-suite, (--define-template-var)
+        Define a Cylc Jinja2/EmPy template variable.
+      -D, --define
+        Define a Rose configuration.
+      -O, --opt-conf-key
+        Define a Rose optional configuration key.
+    Env:
+      ROSE_SUITE_OPT_CONF_KEYS
+        Define a set of Rose optional configuration keys to append to the list.
+```
+
+## Proposed Changes
+
+1. The `rose-suite.conf` file should be installed into the run directory.
+   * For consistency and clarity.
+   * To enable the configuration to be inspected with standard tooling
+     (`rose config`).
+2. Configuration alterations (via CLI or environment) should be written
+   to the filesystem so that subsequent Cylc commands
+   (e.g. `play`, `get-config`, `validate`)
+   have access to the required information (i.e. template variables).
+2. `cylc reinstall` should remember previously defined configuration
+   alterations.
+   * This means we can reload and restart flows safely without having
+     to remember all previously specified options.
+
+## Proposed Implementation
+
+Meddling with the Rose configuration during re-installation is likely to be
+a rare occurrence and what follows is somewhat overblown for purpose, however,
+should reduce to far less code than text in this proposal.
+
+### 1. Any new "defines" should be written to a new optional configuration with a special protected name
+
+Using an optional configuration makes good sense as it would work natively with
```

Comment:
I think that in addition to a protected name the file should contain a comment at the top warning the user that fiddling with it may produce effects that the developer hasn't planned for?

Reply:
Er, the user is free to fiddle with this file (and will be able to do so via Rose Edit!)

Could add a comment to say that changes will only take effect on restart.

Reply:
We could have a comment explaining what the file is?
As for a warning, that applies to editing any installed files, not just this one.

> Could add a comment to say that changes will only take effect on reinstall (as opposed to restart).

Really? Surely this optional config is read on restart, reload, etc in the same way as `cylc.flow`?

Reply:
The `!opts` setting only takes effect on reinstall.